src/server/mod.rs

```diff
@@ -80,75 +66,29 @@ where
     }
 
     /// Serve the `io` stream via default hyper http settings
-    pub fn serve<I>(&mut self, io: I) -> Serve<S, B, I>
+    pub fn serve<I>(&mut self, io: I) -> Serve<S::MakeError>
     where
-        I: AsyncRead + AsyncWrite + 'static,
+        I: AsyncRead + AsyncWrite + Send + 'static,
     {
-        let http = Http::new().with_executor(DefaultExecutor::current());
+        let http = Http::new();
         self.serve_with(io, http)
     }
 
     /// Serve the `io` stream via the provided hyper http settings
-    pub fn serve_with<I>(&mut self, io: I, http: Http<DefaultExecutor>) -> Serve<S, B, I>
+    pub fn serve_with<I>(&mut self, io: I, http: Http) -> Serve<S::MakeError>
```

Comment:
How about `-> impl Future<Item = (), Error = Error<S::MakeError>> + Send + 'static`?

Reply:
I generally prefer to not have `impl Trait` in a library that exposes futures as it doesn't allow you to name the type without existential types.

Reply:
Ok, better API is more important. Thank you!